File: test/language/expressions/java/expressions/UsedBeforeTypeTest.java
package expressions;

import static org.junit.Assert.*;

import org.junit.Test;

import expressions.used_before_type.Color;
import expressions.used_before_type.UsedBeforeTypeExpression;

public class UsedBeforeTypeTest
{
    @Test
    public void bitSizeOfWithOptional()
    {
        final UsedBeforeTypeExpression usedBeforeTypeExpression = new UsedBeforeTypeExpression(Color.RED, true);

        assertEquals(USED_BEFORE_TYPE_EXPRESSION_BIT_SIZE_WITH_OPTIONAL, usedBeforeTypeExpression.bitSizeOf());
    }

    @Test
    public void bitSizeOfWithoutOptional()
    {
        final UsedBeforeTypeExpression usedBeforeTypeExpression = new UsedBeforeTypeExpression();
        usedBeforeTypeExpression.setColor(Color.BLUE);

        assertEquals(USED_BEFORE_TYPE_EXPRESSION_BIT_SIZE_WITHOUT_OPTIONAL,
                usedBeforeTypeExpression.bitSizeOf());
    }

    private static final int USED_BEFORE_TYPE_EXPRESSION_BIT_SIZE_WITH_OPTIONAL = 8;
    private static final int USED_BEFORE_TYPE_EXPRESSION_BIT_SIZE_WITHOUT_OPTIONAL = 7;
}

File: test/extensions/language/expressions/python/UsedBeforeTypeTest.py
import Expressions


class UsedBeforeTypeTest(Expressions.TestCase):
    def testBitSizeOfWithOptional(self):
        usedBeforeTypeExpression = self.api.UsedBeforeTypeExpression(self.api.Color.RED, True)
        self.assertEqual(
            self.USED_BEFORE_TYPE_EXPRESSION_BIT_SIZE_WITH_OPTIONAL, usedBeforeTypeExpression.bitsizeof()
        )

    def testBitSizeOfWithoutOptional(self):
        usedBeforeTypeExpression = self.api.UsedBeforeTypeExpression()
        usedBeforeTypeExpression.color = self.api.Color.BLUE
        self.assertEqual(
            self.USED_BEFORE_TYPE_EXPRESSION_BIT_SIZE_WITHOUT_OPTIONAL, usedBeforeTypeExpression.bitsizeof()
        )

    USED_BEFORE_TYPE_EXPRESSION_BIT_SIZE_WITH_OPTIONAL = 8
    USED_BEFORE_TYPE_EXPRESSION_BIT_SIZE_WITHOUT_OPTIONAL = 7
